### DISCUSSION TITLE
Search by note title instead of note filename

### DISCUSSION THREAD

**Discussion by darlal** [darlal/obsidian-switcher-plus#20] [Ideas]
A new "title" mode could be implemented that would allow you to search, display, and navigate to different notes using their title instead of their filename. The title could be indexed as the first H1 contained in the note. For example, if you use the Zettelkasten method, then all note filenames would be numerical IDs so being able to search via title would be useful.

**Comment by bobheadxi** [comment]
I'd be interested in expanding the "scope" of a search, e.g. shortcuts to include headings, then even drop into block search 😁

Sometimes I find myself wanting to quick switch into a note, but can't quite remember the name, in which case I might want to search through headings, then just through all the content to find the block I want.

**Reply by darlal** [reply]
Thanks for the comment, that's kind of what I'm thinking with the headings. Right now, you have to already know the filename that contains the thing you want, then you can use symbol search to navigate to it (including headers). This would enable you to go straight to headers without first identifying file.

Blocks are interesting in that I think you'd still want to start with a file, then filter content by blocks, similar to how Symbols search is done currently.

**Comment by darlal** [comment]
@bobheadxi the ability to search through headings is now live with release [1.0.2](https://github.com/darlal/obsidian-switcher-plus/releases/tag/1.0.2). Check it out. 

I will close this thread out. Thanks.
